src/runtime/components/NuxtImg.vue

```diff
@@ -1,18 +1,24 @@
 <template>
   <img
+    v-show="isImageLoaded || !$slots.placeholder"
     ref="imgEl"
-    :class="props.placeholder && !placeholderLoaded ? props.placeholderClass : undefined"
+    :class="props.placeholder && !isImageLoaded ? props.placeholderClass : undefined"
     v-bind="{
       ...isServer ? { onerror: 'this.setAttribute(\'data-error\', 1)' } : {},
       ...imgAttrs,
       ...attrs,
     }"
     :src="src"
   >
+
+  <slot
+    v-if="!isImageLoaded && $slots.placeholder"
```

Comment:
```suggestion
    v-else
```

Reply:
@danielroe I don't think this would work, I used v-show on the img tag so that it always renders on server side.

Reply:
ahhh, quite right

Reply:
seems like this is a vue ssr bug, right? maybe instead use `isImageLoaded`? can we avoid checking `$slots.placeholder`?

Reply:
I am not sure what you mean by Vue SSR bug. I am using v-show because I need the element to be prepared on the server (it renders the <img> with the placeholder src or the desired image, or it renders the slot immediately on the server). If I use v-if, then the onload event won't be triggered because the element doesn't exist in the DOM. We can probably avoid checking $slots.placeholder.

Reply:
I simplified the condition